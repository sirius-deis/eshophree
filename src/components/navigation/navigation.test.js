import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navigation from './navigation';

describe('Navigation component', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should render menu items', () => {
    render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
    );
    expect(screen.getByText(/home/i)).toBeInTheDocument();
    expect(screen.getByText(/categories/i)).toBeInTheDocument();
    expect(screen.getByText(/shop/i)).toBeInTheDocument();
    expect(screen.getByText(/about/i)).toBeInTheDocument();
    expect(screen.getByText(/blog/i)).toBeInTheDocument();
    expect(screen.getByText(/contact/i)).toBeInTheDocument();
  });
});
