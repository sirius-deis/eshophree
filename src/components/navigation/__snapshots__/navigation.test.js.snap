// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should match snapshot 1`] = `
<div>
  <nav
    class="sc-bgqQcB cbBJqk"
  >
    <div
      class="container"
    >
      <ul
        class="sc-gTRrQi epUtYC"
        style="font-size: 1.7rem;"
      >
        <li
          class="sc-ewnqHT iWesBX"
        >
          <a
            href="/home"
          >
            home
          </a>
        </li>
        <li
          class="sc-ewnqHT iWesBX"
        >
          <a
            href="/categories"
          >
            categories
          </a>
        </li>
        <li
          class="sc-ewnqHT iWesBX"
        >
          <a
            href="/shop"
          >
            shop
          </a>
        </li>
        <li
          class="sc-ewnqHT iWesBX"
        >
          <a
            href="/about"
          >
            about
          </a>
        </li>
        <li
          class="sc-ewnqHT iWesBX"
        >
          <a
            href="/blog"
          >
            blog
          </a>
        </li>
        <li
          class="sc-ewnqHT iWesBX"
        >
          <a
            href="/contact"
          >
            contact
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
